app/helpers/maps_helper.rb

```diff
@@ -1,19 +1,12 @@
 module MapsHelper
   def vacancy_map_markers(vacancies, polygon, coordinates, radius)
-    organisation_ids = Organisation.in_vacancy_ids(vacancies.pluck(:id))
-                                   .within_polygon(polygon)
-                                   .within_area(coordinates, radius)
-                                   .pluck(:id)
-
-    vacancies.map { |vacancy|
-      vacancy.organisations.select { |organisation| organisation.id.in?(organisation_ids) }.map do |organisation|
-        {
-          id: vacancy.id,
-          parent_id: organisation.id,
-          geopoint: RGeo::GeoJSON.encode(organisation.geopoint)&.to_json,
-        }
-      end
-    }.flatten
+    Marker.live.within_polygon(polygon).within_area(coordinates, radius).where(vacancy_id: vacancies.pluck(:id)).map do |marker|
```

Comment:
I reckon this can be in the `VacancySearch` service - so instead of that fetching _all_ matching vacancies in the service, passing them to the view, and then into the helper, only to use their ID to fetch the markers, we could have something like this in `VacancySearch`:

```ruby
def markers
  @markers ||= Marker.live.search_by_location(location, radius).pluck(:vacancy_id, :organisation_id, :geopoint)
end
```